atdgen.ml

```diff
@@ -273,11 +273,13 @@ end
 
 let make_abstract ((_loc,annot),init) types =
   let esgg_section_name = "esgg" in
-  let module_name = Atd.Annot.get_opt_field ~parse:(fun s -> Some s) ~sections:[esgg_section_name] ~field:"from" annot in
-  let annot =
-    annot
-    |> List.filter (fun (section_name, _) -> section_name <> esgg_section_name)
-    |> Atd.Annot.set_field ~loc ~section:"ocaml" ~field:"from" module_name
+  let module_name_opt = Atd.Annot.get_opt_field ~parse:(fun s -> Some s) ~sections:[esgg_section_name] ~field:"from" annot in
+  let annot = 
+    let annot' = match module_name_opt with
+    | Some module_name ->  Atd.Annot.set_field ~loc ~section:"ocaml" ~field:"from" (Some module_name) annot
```

Comment:
This unwrapping and wrapping of `module_name` is necessary because annotations can have `None` as a value, e.g. `<ocaml from>` vs `<ocaml from="...">`. In this case, we don't want the former to happen

Reply:
this should be comment in code ^